### DISCUSSION TITLE
Proposed Dataset for Lesson 84. Creating a Straight Line Dataset to See if Our Model is Learning Anything

### DISCUSSION THREAD

**Discussion by ChrisLeeSG** [mrdbourke/pytorch-deep-learning#1081] [Ideas]
First of all, I wish you thank you for putting together this course. I've always wanted to learn neural network and realizing my dream through this course.

*84. Creating a Straight Line Dataset to See if Our Model is Learning Anything*

## Idea / Proposal

Use a linear classification dataset so that we can truly test if our model is capable of performing binary classification. 

## Benefit

Using this dataset, we don't have to make any changes to our model, loss function and optimizer when we are testing if our model can learn anything.

## How

Create a straight line and classify randomly generated points above the line as 1.0 and points below the line as 0.0.

## My Code

```python
import torch 
from torch import nn

# Assign weight and bias for a straight line
weight = 0.8
bias = 0.1

# Randomly assign X1 and X2 values
X = torch.rand(1000, 2)

# Calculate line using the equation: line = weight*X1 + bias
line = weight*X[:, 0] + bias

# For y, assign  0 if X2 is below line and 1 otherwise
y = X[:, 1] - line
y = torch.where(y < 0, torch.tensor(0.0), torch.tensor(1.0))
```
## Result

Using this dataset to train and test our model, proves that it can perform a binary classification on this dataset.

<img width="493" alt="MatPlotLib" src="https://github.com/user-attachments/assets/ce5214b7-68ad-490c-8d4f-cc431f0d0eee">
